client/state/jetpack-connect/reducer/jetpack-auth-attempts.js

```diff
@@ -6,26 +6,31 @@ import { AUTH_ATTEMPS_TTL } from '../constants';
 import { isStale } from '../utils';
 import { JETPACK_CONNECT_COMPLETE_FLOW, JETPACK_CONNECT_RETRY_AUTH } from 'state/action-types';
 import { jetpackAuthAttemptsSchema } from './schema';
+import { keyedReducer } from 'state/utils';
 
-export default function jetpackAuthAttempts( state = {}, action ) {
-	switch ( action.type ) {
-		case JETPACK_CONNECT_RETRY_AUTH:
-			const slug = action.slug;
-			let currentTimestamp = state[ slug ] ? state[ slug ].timestamp || Date.now() : Date.now();
-			let attemptNumber = action.attemptNumber;
-			if ( attemptNumber > 0 ) {
-				const now = Date.now();
-				if ( isStale( currentTimestamp, AUTH_ATTEMPS_TTL ) ) {
-					currentTimestamp = now;
-					attemptNumber = 0;
+const jetpackAuthAttempts = keyedReducer(
+	'slug',
+	( state = undefined, { type, attemptNumber } ) => {
```

Comment:
I don't see any reason to pass the attempt number into this reducer. It can and should just increment the one in state.